pkg/console/operator/sync_v400.go

```diff
@@ -392,6 +403,86 @@ func (co *consoleOperator) SyncRoute(operatorConfig *operatorv1.Console) (*route
 	return rt, rtIsNew, rtErr
 }
 
+func (c *consoleOperator) SyncCustomLogoConfigMap(operatorConfig *operatorsv1.Console) error {
+	// validate first, to avoid a broken volume mount & a crashlooping console
+	okToMount, err := c.ValidateCustomLogo(operatorConfig)
+
+	if okToMount {
+		if err := c.UpdateCustomLogoSyncSource(operatorConfig); err != nil {
+			return err
+		}
+	}
+	return err
+}
+
+// on each pass of the operator sync loop, we need to check the
```

Comment:
Leaving the comment in here as there is a bit to understand via the resource syncer.  The resource syncer is a separate controller with its own sync loop. It watches all the configmaps in `openshift-config`, and checks the `operator` config. If the `operator` config has a named configmap, it will sync ONLY that configmap into the `console` namespace.

Then, the main sync loop of the operator will evaluate the `operator` config & the `custom-logo` configmap to a reasonable degree & will mount the logo if it checks out.

Reply:
When we call this func we are:
- telling it to sync `foo-configmap` from `openshift-config` as `custom-logo` in `openshift-console`
- telling it to update, `bar-configmap` now should be sync'd instead
- telling it to eliminate the configmap if it is no longer being used.